creator/analyses/file_types.py

```diff
@@ -50,18 +72,27 @@
         ],
         "template": "family_trio_config.py",
     },
+    "FCM": {
+        "name": "Complex Family",
+        "required_columns": [
+            "Source Participant ID",
+            "Target Participant ID",
+            "Family Relationship",
+        ],
+        "template": "complex_family_config.py",
+    },
     "GWO": {
         "name": "Genomic Workflow Output Manifest",
         "required_columns": [
-            'Cavatica ID',
-            'Cavatica Task ID',
-            'KF Biospecimen ID',
-            'KF Participant ID',
-            'KF Family ID',
-            'Filepath',
-            'Data Type',
-            'Workflow Type',
-            'Source Read'
+            "Cavatica ID",
+            "Cavatica Task ID",
+            "KF Biospecimen ID",
+            "KF Participant ID",
+            "KF Family ID",
+            "Filepath",
+            "Data Type",
+            "Workflow Type",
+            "Source Read",
```

Comment:
Per my comment on notion, most of these fields are useless for ingesting and the useful parts are harder to actually use than necessary (partly a consequence of the mandate to explicitly link SE information from source files to computed files). 

Right now we need to link these GFs to their parent BSes and SEes. Let's imagine a hypothetical future where we either also/instead link to their parent GFs (right now we can ad-hoc use the GF metadata field for that).

If you can get Allison to say that we don't need to make SEGFs for output files (please try because doing it adds a large procedural burden), then we would only need filepath, biospecimen KFIDs (if linking specimens to GFs like we do now), and source files (if linking source GFs to output GFs which we don't do now).

Otherwise the next best option would be to have filepath, source Sequencing Experiment KFIDs, biospecimen KFIDs (if linking specimens to GFs like we do now), and source files (if linking source GFs to output GFs which we don't do now), since those are the things that we have to link.

And if you can't get Sequencing Experiment KFIDs, then you either want biospecimen external IDs instead of KFIDs so that you can put them into the main ingest package and automatically get the SE relationship assignment from the original source data or you have to do an additional round of query processing to discover the SEs from the source GFs before you can ingest the new files.

Reply:
These files are generated automatically by a script from the BIX team, so right now they are all marked as required, even it's not the case.

Reply:
I recognize that one purpose of these templates is to handle what we know exists, but another purpose is also to guide what we get. I think this effort should include a conversation with Allison and BIX about what values are most useful.